### PR TITLE
[6.4.x][RHBRMS-2447] fix asset-mgmt-project kjar to work OOTB

### PR DESCRIPTION
**Pull request by psiroky**
* before this change, WB logs WARNing that it can not
   find the asset-mgmt kjar, and then successfully installs
   it itself (so the WARN was pretty misleading)

 * the kjar is now built via maven-invoker-plugin and without
   any parent. The filtering avoids hardcoded any versions.

Depends on https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/216.

@manstis, @Salaboy, @nmirasch could one of you please review this?

### CODE REVIEW

**Comment by psiroky** [comment]
@ryanzhang, @AntStephenson would be great to get some feedback from you guys. Using the maven-invoker-plugin is not really conventional. Could that cause some unforeseen issue during the prod. build?

File: guvnor-asset-mgmt/guvnor-asset-mgmt-project/pom.xml [hunk 1]
```diff
@@ -4,40 +4,76 @@
...
   </parent>
 
   <artifactId>guvnor-asset-mgmt-project</artifactId>
-  <packaging>jar</packaging>
+  <packaging>pom</packaging>
 
-  <name>Guvnor - Asset Management project</name>
-  <description>Guvnor - Asset management project (includes processes to manage business assets)</description>
+  <name>Guvnor - Asset Management Project</name>
+  <description>Guvnor - Asset management Project (includes processes to manage business assets)</description>
```
Comment (manstis) [inline guvnor-asset-mgmt/guvnor-asset-mgmt-project/pom.xml]: If I was nit picking I'd say you should have a capital "M" for [M]anagement ;)

File: guvnor-asset-mgmt/guvnor-asset-mgmt-project/src/main/filtered-resources/kjar/pom.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (manstis) [inline guvnor-asset-mgmt/guvnor-asset-mgmt-project/src/main/filtered-resources/kjar/pom.xml]: Does this need a license?

File: guvnor-asset-mgmt/guvnor-asset-mgmt-project/src/main/filtered-resources/kjar/pom.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>guvnor-asset-mgmt-project</artifactId>
+  <groupId>org.guvnor</groupId>
+  <version>${version.org.kie}</version>
+
+  <packaging>jar</packaging>
+
+  <name>Guvnor - Asset Management project</name>
```
Comment (manstis) [inline guvnor-asset-mgmt/guvnor-asset-mgmt-project/src/main/filtered-resources/kjar/pom.xml]: Capitals.. (I only mention it as I see you went to the trouble of fixing elsewhere)

File: guvnor-asset-mgmt/guvnor-asset-mgmt-project/src/main/filtered-resources/kjar/pom.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  <description>Guvnor - Asset management project (includes processes to manage business assets)</description>
```
Comment (manstis) [inline guvnor-asset-mgmt/guvnor-asset-mgmt-project/src/main/filtered-resources/kjar/pom.xml]: Capitals.. (I only mention it as I see you went to the trouble of fixing elsewhere)

File: guvnor-asset-mgmt/guvnor-asset-mgmt-project/pom.xml (hunk 1 above)
Comment (psiroky) [inline guvnor-asset-mgmt/guvnor-asset-mgmt-project/pom.xml]: You are correct, will fix that.

**Comment by manstis** [comment]
@psiroky Minor cosmetic comments (I see you've been doing similar on PRs, so thought you'd like to know).. otherwise looks OK to me (I trust that the resulting JAR doesn't have a parent pom).

File: guvnor-asset-mgmt/guvnor-asset-mgmt-project/src/main/filtered-resources/kjar/pom.xml (hunk 2 above)
Comment (psiroky) [inline guvnor-asset-mgmt/guvnor-asset-mgmt-project/src/main/filtered-resources/kjar/pom.xml]: I _think_ it does not. We are generally not including headers in pom.xml files (even though this one is a bit different for sure).

File: guvnor-asset-mgmt/guvnor-asset-mgmt-project/src/main/filtered-resources/kjar/pom.xml (hunk 3 above)
Comment (psiroky) [inline guvnor-asset-mgmt/guvnor-asset-mgmt-project/src/main/filtered-resources/kjar/pom.xml]: Damn, I should have first fix it and then do the copy, not the other way around. Thanks, will fix this one as well.

**Comment by psiroky** [comment]
@manstis this solution is not exactly ideal, but it is the best I could think of. I tried to deploy the kie-wb WAR with the change -- and no WARN or ERRORs were logged, so I am assuming it is good.

**Comment by manstis** [comment]
@psiroky Presumably you tested that with your (applicable) local M2 cleared?

**Comment by psiroky** [comment]
@manstis yes, I removed all the asset-mgmt related jars before starting the EAP.

**Comment by manstis** [comment]
+1 to merge (but will wait for others to comment before proceeding).

**Comment by ryanzhang** [comment]
+1.
This looks ok from my pespective. we will import the new plugin. 
Run a local build in our product tooling, it built ok too.

**Comment by manstis** [comment]
**DO NOT MERGE YET**

**Comment by psiroky** [comment]
jenkins retest this

**Pull request merged**